0012-species-data.md

```diff
@@ -0,0 +1,142 @@
+    the admin panel is overwhelming, and if one of their species has
+    not yet been examined by a GS botanist, the organization cannot
+    select an identification. In addition, for the botanist to have
+    access to an organization’s local species list is very helpful,
+    but communicating with the organization to get this list can be
+    hard.  What is needed is: a _short_ list of local names that the
+    organization can edit and add to.  
+ 2. The current way that determination decisions are recorded in the
+    DB prevents many important queries. The act of choosing a species
+    is a key data element, and needs its own table.
+
+## Proposed solution
+
+**Overall**: _1. Separate local, organization species from global,
+vetted species, and 2. track each determination event, and each
+local-to-global mapping within the database._
+
+### User workflow
+
+ 1. Each org creates a set of local names in the admin panel: name entries
+    could just be local language names, or they may also have a
+    tentative, unvetted scientific name. Those are the only names that
+    appear to the org user (who starts with a blank list of names).
+ 2. The org user then determines the local name for each capture. The
+    determinations are logged in a joining `det` table, so that
+    multiple opinions can be made about a tree’s name. In general, any
+    display of the name will use the latest determination.
+ 3. A ‘species admin’ level user of the admin panel will create
+    mappings between the localname (= morphotype) and a vetted list of
+    global names (these may also just be morphotype codes in the case
+    where the final scientific name has not been settled). E.g.,
+    `"mangga"@orgX` is mapped to global `MANGINDI`. Once
+    this vetting has happened, individual trees can are then linked
+    out to the global knowledge base. E.g., wood densities can be
+    imported and carbon calculations could thus be made.
+
+### Modified schema
+
+```
+            (NEW)               (NEW)           (NEW)             
+trees       det                 localname       taxonmap            tree_species
+=====       ==========          =========       =============       ============
+id    <-+   id              +-> id        <-+   id              +-> id
+        +-- tree_id*        |   name        +-- localname_id*   |   name
+            localname_id* --+   sci_name%       species_id*   --+
+            user_id*            org_id*         user_id*
+            timestamp           user_id*        confidence
+                                timestamp       notes
+                                                timestamp
+```
+
+Notes: 
+
+ * A “*” indicates a foreign key; a “%” indicates a field may be NULL.
+ * `user_id` joins to existing `admin_user.id`.
+ * `org_id` joins to existing `entity.id` for an organization.
+ * The `trees` table listed here refers to the existing schema where a
```

Comment:
This will indeed be implemented in the new domain model where we separate `captures` from `trees`. My feeling is that _captures_ will have changeable species determinations (used, in part, to aid with capture matching), but trees will have a single `species_id` field.

In your schema, `trees` becomes `capture` and `det.tree_id` becomes `det.capture_id`.

The natural place for `localname`, `taxonmap` and `tree_species` is a new `species` or `herbarium` microservice, but I think `det` would be best placed in the `treetracker` schema, since it relates to tracking/tagging events.

Reply:
Thanks for the review!

> My feeling is that captures will have changeable species determinations (used, in part, to aid with capture matching), but trees will have a single species_id field.

I think it is important that species information is _not_ attached to trees in two places, i.e. via `capture -> det -> localname -> taxonmap -> species` and also with a single`tree.species_id` field. The information in `tree.species_id` should be lookup information from the most recent determination. Otherwise the DB is not normalized for species info.

Reply:
> The natural place for localname, taxonmap and tree_species is a new species or herbarium microservice, but I think det would be best placed in the treetracker schema, since it relates to tracking/tagging events.

Seems like a good classification. But, just to be sure I understand... these separate microservice and treetracker schemas would still exist in the same database, and all tables would still be able to be accessed via a single SQL query? That is vital for data analysis and herbarium work.

Reply:
> these separate microservice and treetracker schemas would still exist in the same database, and all tables would still be able to be accessed via a single SQL query?
Correct